py3status/modules/mpris.py

```diff
@@ -0,0 +1,497 @@
+        self._loop.run()
+
+    def _start_listener(self):
+        self._dbus.con.signal_subscribe(
+            None,
+            'org.freedesktop.DBus',
+            'NameOwnerChanged',
+            None,
+            None,
+            0,
+            self._on_available_players_changed,
+        )
+        t = Thread(target=self._start_loop)
+        t.daemon = True
+        t.start()
+
+    def _update_metadata(self, metadata=None):
+        is_stream = False
+
+        if metadata is None:
+            metadata = self._player.Metadata
+
+        try:
+            metadata = self._player.Metadata
+            if len(metadata) > 0:
+                url = metadata.get('xesam:url')
+                is_stream = url is not None and 'file://' not in url
+                self._data['title'] = metadata.get('xesam:title')
+                self._data['album'] = metadata.get('xesam:album')
+
+                if metadata.get('xesam:artist') is not None:
+                    self._data['artist'] = metadata.get('xesam:artist')[0]
+                else:
+                    # we assume here that we playing a video and these types of
+                    # media we handle just like streams
+                    is_stream = True
+
+                length_ms = metadata.get('mpris:length')
+                if length_ms is not None:
+                    self._data['length'] = _get_time_str(length_ms)
+            else:
+                # use stream format if no metadata is available
+                is_stream = True
+        except Exception:
+            self._data['error_occurred'] = True
+
+        if is_stream and self._data['title']:
+            # delete the file extension
+            self._data['title'] = re.sub(r'\....$', '', self._data['title'])
+
+    def mpris(self, i3s_output_list, i3s_config):
+        """
+        Get the current output format and return it.
+        """
+        cached_until = time() + self.cache_timeout
+
+        if self._dbus is None:
+            self._dbus = SessionBus()
+            self._start_listener()
+
+        running_player = self._get_highest_prioritized()
+        if self._player is None or self._player_name != running_player:
+            # TODO: This should also works with "del self._player"
+            if self._player_subscription:
+                self._player_subscription.disconnect()
+            self._player_name = running_player
+            self._player = self._get_player(running_player)
+            self._init_data()
+            if self._player is not None:
+                self._player_subscription = self._player.PropertiesChanged.connect(self._on_change)
+
+        if self._player is None:
+            self._player = None
+            self._player_name = 'None'
+            self._player_subscription = None
+            text = self.format_none
+            color = i3s_config['color_bad']
+            composite = ({
+                'full_text': text,
+                'color': color,
+            })
```

Comment:
a composite is a list so `composite = [{ ... {[` you have an error when no player.